### PR TITLE
Html

### PR DESCRIPTION
**Pull request by cloudwerewolf**

### CODE REVIEW

File: note.js [hunk 1]
```diff
@@ -1,23 +1,84 @@
...
+        xBtn.value = "❌";
+        xBtn.id = notes + "btn";
+        xBtn.noteid = notes;
+        xBtn.onmouseup = function() {delNote(this.noteid);};
+        xBtn.style.visibility = "hidden";
+        notespan.appendChild(xBtn);
+        var shrBtn = document.createElement("input");
+        shrBtn.type = "Submit";
+        shrBtn.className = "share";
+        shrBtn.value = "⤿";
```
Comment (raymeskhoury) [inline note.js]: This is a bit disproportionate to the X. Maybe ➡ instead?

File: note.js [hunk 2]
```diff
@@ -1,23 +1,84 @@
...
+        var xBtn = document.createElement("input");
+        xBtn.type = "Submit";
+        xBtn.className = "delBtn";
+        xBtn.value = "❌";
+        xBtn.id = notes + "btn";
+        xBtn.noteid = notes;
+        xBtn.onmouseup = function() {delNote(this.noteid);};
+        xBtn.style.visibility = "hidden";
+        notespan.appendChild(xBtn);
+        var shrBtn = document.createElement("input");
```
Comment (raymeskhoury) [inline note.js]: We should only add this button if navigator.share is available

File: note.js [hunk 3]
```diff
@@ -1,23 +1,84 @@
...
+    var note = document.getElementById(id);
+    var notediv = note.parentNode;
+    var div = document.getElementById("notepage");
+    div.removeChild(notediv);
+    notelist.splice(id, 1);
+}
+function shareNote(id) {
+    var notes = document.getElementById(id);
+    if (typeof navigator.share !== 'undefined') {
+        event.preventDefault();
```
Comment (raymeskhoury) [inline note.js]: Why is this needed?

File: note.js [hunk 4]
```diff
@@ -1,23 +1,84 @@
...
+        navigator.share({ title: "NotePen Note", text: notes.value })
+        .catch(function() {alert("Sorry, sharing is not supported by this browser.")});
```
Comment (raymeskhoury) [inline note.js]: alert() isn't really a good UI experience. If we don't add the button when navigator.share isn't available, this probably won't be hit in most cases.

File: note.js [hunk 5]
```diff
@@ -1,23 +1,84 @@
...
     var div = document.getElementById("notepage");
-    for (var notes in retnotelist){
+    for (var notes in retnotelist) {
+        var notediv = document.createElement("div");
+        notediv.className = "noteholder";
+        div.appendChild(notediv);
+        var notespan = document.createElement("span");
+        notespan.className = "notebuttons";
+        notediv.appendChild(notespan);
         var text = document.createElement("textarea");
```
Comment (raymeskhoury) [inline note.js]: let's make this non-editable

File: note.js [hunk 6]
```diff
@@ -1,23 +1,84 @@
...
+}
+function shareNote(id) {
+    var notes = document.getElementById(id);
+    if (typeof navigator.share !== 'undefined') {
+        event.preventDefault();
+        navigator.share({ title: "NotePen Note", text: notes.value })
+        .catch(function() {alert("Sorry, sharing is not supported by this browser.")});
+    }
+}
+function showBtn(id) {
```
Comment (raymeskhoury) [inline note.js]: When this is called we should clear the timeouts that are set in fade(). Otherwise you will get weird effects when you press it multiple times.

File: note.js [hunk 7]
```diff
@@ -1,23 +1,84 @@
...
+        var notespan = document.createElement("span");
+        notespan.className = "notebuttons";
+        notediv.appendChild(notespan);
         var text = document.createElement("textarea");
         text.name = "notes"
         text.cols = 45;
         text.rows = 5;
         text.className = "notelist";
-        this.id = notes;
+        text.onfocus = function() {showBtn(this.id);};
```
Comment (raymeskhoury) [inline note.js]: I think we should call showBtn in the onmousemove handler as well. That will mean that when you move your mouse over it on desktop it still shows up.

File: note.js (hunk 6 above)
Comment (cloudwerewolf) [reply note.js]: Isn't that what's being done in line 6?
